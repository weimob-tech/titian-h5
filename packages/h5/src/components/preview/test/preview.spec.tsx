import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiPreview } from '../index';

describe('渲染 ti-preview', () => {
  it('基本渲染', async () => {
    const page = await newSpecPage({
      components: [TiPreview],
      template: () => <ti-preview></ti-preview>,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('隐藏页码', async () => {
    const page = await newSpecPage({
      components: [TiPreview],
      template: () => <ti-preview displayNumber={false}></ti-preview>,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('隐藏标题', async () => {
    const page = await newSpecPage({
      components: [TiPreview],
      template: () => <ti-preview displayTitle={false}></ti-preview>,
    });
    expect(page.root).toMatchSnapshot();
  });
});
