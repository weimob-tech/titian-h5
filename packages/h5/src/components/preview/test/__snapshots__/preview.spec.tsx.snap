// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-preview 基本渲染 1`] = `
<ti-preview>
  <mock:shadow-root>
    <ti-transition class="titian-preview titian-preview-mask" destroy-on-exit="" name="fade" style="--protected-preview-z-index: 11000;">
      <div class="titian-preview-file-wrap">
        <ti-swiper current="0"></ti-swiper>
      </div>
      <div class="titian-preview-content-box">
        <div></div>
        <div class="titian-preview-serial-number">
          1/0
        </div>
      </div>
    </ti-transition>
  </mock:shadow-root>
</ti-preview>
`;

exports[`渲染 ti-preview 隐藏标题 1`] = `
<ti-preview>
  <mock:shadow-root>
    <ti-transition class="titian-preview titian-preview-mask" destroy-on-exit="" name="fade" style="--protected-preview-z-index: 11000;">
      <div class="titian-preview-file-wrap">
        <ti-swiper current="0"></ti-swiper>
      </div>
      <div class="titian-preview-content-box">
        <div class="titian-preview-serial-number">
          1/0
        </div>
      </div>
    </ti-transition>
  </mock:shadow-root>
</ti-preview>
`;

exports[`渲染 ti-preview 隐藏页码 1`] = `
<ti-preview>
  <mock:shadow-root>
    <ti-transition class="titian-preview titian-preview-mask" destroy-on-exit="" name="fade" style="--protected-preview-z-index: 11000;">
      <div class="titian-preview-file-wrap">
        <ti-swiper current="0"></ti-swiper>
      </div>
      <div class="titian-preview-content-box">
        <div></div>
      </div>
    </ti-transition>
  </mock:shadow-root>
</ti-preview>
`;
